meilisearch/index.py

```diff
@@ -754,6 +754,33 @@ def delete_documents(self, ids: List[Union[str, int]]) -> TaskInfo:
         )
         return TaskInfo(**response)
 
+    def delete_documents_by_filter(
+        self, filter: Union[str, List[Union[str, List[str]]]]  # pylint: disable=redefined-builtin
```

Comment:
Pylint doesn't like the parameter being named `filter` so I set it to ignore this error. If you would rather call it something else I can do that. If you want to rename do you have any suggestions for a name? I couldn't think of a good one.

Reply:
For me, it's ok to keep `filter`. Is this the same issue with `filters` in the plural?
I don't think we should give it any other name than the one it has

Reply:
pylint would be ok with `filters`. The reason pylint doesn't like `filter` is because python has a built in `filter` function so it is trying to prevent overlap. In this case I can't think of a time where naming it `filter` would actually cause any issues though so I think either ignoring the pylint error or renaming to `filters` are both fine options, which ever you prefer.